components/core/src/Query.hpp

```diff
@@ -172,7 +172,7 @@ class Query {
     void set_ignore_case(bool ignore_case) { m_ignore_case = ignore_case; }
 
     void set_search_string(std::string const& search_string);
-    void add_sub_query(SubQuery const& sub_query);
+    void add_sub_queries(std::vector<SubQuery>& sub_query);
```

Comment:
* Let's remove the `&` so that a caller could pass in an rvalue.
* How about calling this method `set_sub_queries`? `add_sub_queries` implies that we're adding sub-queries to any sub-queries already in `m_sub_queries`.

Reply:
But it will introduce a copy?